hydromt/data_adapter/rasterdataset.py

```diff
@@ -578,70 +585,114 @@ def _set_metadata(self, ds):
+            with different unit than dst_crs
         """
-        # common pyproj crs axis units
-        known_units = ["degree", "metre", "US survey foot"]
-        if self.zoom_levels is None or len(self.zoom_levels) == 0:
-            logger.warning("No zoom levels available, default to zero")
-            return 0
-        zls = list(self.zoom_levels.keys())
-        if zoom_level is None:  # return first zoomlevel (assume these are ordered)
-            return next(iter(zls))
-        # parse zoom_level argument
-        if (
+        # check zoom level
+        zls_dict = self.zoom_levels if zls_dict is None else zls_dict
+        dst_crs = self.crs if dst_crs is None else dst_crs
+        if zls_dict is None or len(zls_dict) == 0 or zoom_level is None:
+            return None
+        elif isinstance(zoom_level, int):
+            if zoom_level not in zls_dict:
+                raise ValueError(
+                    f"Zoom level {zoom_level} not defined."
+                    f"Select from {list(zls_dict.keys())}."
+                )
+            zl = zoom_level
+            dst_res = zls_dict[zoom_level]
+        elif (
             isinstance(zoom_level, tuple)
             and isinstance(zoom_level[0], (int, float))
             and isinstance(zoom_level[1], str)
             and len(zoom_level) == 2
+            and dst_crs is not None
         ):
-            res, unit = zoom_level
-            # covert 'meter' and foot to official pyproj units
-            unit = {"meter": "metre", "foot": "US survey foot"}.get(unit, unit)
-            if unit not in known_units:
-                raise TypeError(
-                    f"zoom_level unit {unit} not understood;"
-                    f" should be one of {known_units}"
-                )
-        elif not isinstance(zoom_level, int):
-            raise TypeError(
-                f"zoom_level argument not understood: {zoom_level}; should be a float"
-            )
-        else:
-            return zoom_level
-        if self.crs:
+            src_res, src_res_unit = zoom_level
             # convert res if different unit than crs
-            crs = pyproj.CRS.from_user_input(self.crs)
-            crs_unit = crs.axis_info[0].unit_name
-            if crs_unit != unit and crs_unit not in known_units:
-                raise NotImplementedError(
-                    f"no conversion available for {unit} to {crs_unit}"
-                )
-            if unit != crs_unit:
-                lat = 0
-                if bbox is not None:
-                    lat = (bbox[1] + bbox[3]) / 2
-                elif geom is not None:
-                    lat = geom.to_crs(4326).centroid.y.item()
+            dst_crs = pyproj.CRS.from_user_input(dst_crs)
+            dst_crs_unit = dst_crs.axis_info[0].unit_name
+            dst_res = src_res
+            if dst_crs_unit != src_res_unit:
+                known_units = ["degree", "metre", "US survey foot", "meter", "foot"]
```

Comment:
would km also be a common one? or is that one not used often?

Reply:
Not in pyproj crs to my knowledge